twitter/twitter.php

```diff
@@ -1420,22 +1426,22 @@ function twitter_createpost(App $a, $uid, $post, $self, $create_user, $only_exis
 
 	$statustext = $converted["plain"];
 
-	if (is_string($post->place->name)) {
+	if (!empty($post->place->name)) {
 		$postarray["location"] = $post->place->name;
 	}
-	if (is_string($post->place->full_name)) {
+	if (!empty($post->place->full_name)) {
 		$postarray["location"] = $post->place->full_name;
 	}
-	if (is_array($post->geo->coordinates)) {
+	if (!empty($post->geo->coordinates)) {
 		$postarray["coord"] = $post->geo->coordinates[0] . " " . $post->geo->coordinates[1];
 	}
-	if (is_array($post->coordinates->coordinates)) {
+	if (!empty($post->coordinates->coordinates)) {
 		$postarray["coord"] = $post->coordinates->coordinates[1] . " " . $post->coordinates->coordinates[0];
 	}
-	if (is_object($post->retweeted_status)) {
+	if (!empty($post->retweeted_status)) {
 		$retweet = twitter_createpost($a, $uid, $post->retweeted_status, $self, false, false, $noquote);
 
-		$retweet['object'] = $postarray['object'];
+		//$retweet['object'] = $postarray['object'];
```

Comment:
Can you please add a comment about why you left code commented out or remove it entirely?